OpenRA.Mods.Common/Widgets/Logic/AssetBrowserLogic.cs

```diff
@@ -670,5 +704,29 @@ void ClearLoadedAssets()
 			// Just in case we're switching away from a type of asset that forced the music to mute.
 			UnMuteSounds();
 		}
+
+		Widget CreateAssetTypesPanel()
+		{
+			var assetTypesPanel = Ui.LoadWidget("ASSET_TYPES_PANEL", null, new WidgetArgs());
+			var assetTypeTemplate = assetTypesPanel.Get<CheckboxWidget>("ASSET_TYPE_TEMPLATE");
+
+			var allAssetTypes = new[] { AssetType.Sprite, AssetType.Model, AssetType.Audio, AssetType.Video, AssetType.Unknown };
+			foreach (var type in allAssetTypes)
+			{
+				var assetType = (CheckboxWidget)assetTypeTemplate.Clone();
+				assetType.GetText = () => type.ToString();
```

Comment:
Minor nit: `ToString` is called on every text render, so I suggest something like
```suggestion
				var text = type.ToString();
				assetType.GetText = () => text;
```